libraries/plugins/custom_operations/include/graphene/custom_operations/custom_operations_plugin.hpp

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) 2019 oxarbitrage and contributors.
+ *
+ * The MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+#pragma once
+
+#include <graphene/app/plugin.hpp>
+#include <graphene/chain/database.hpp>
+
+#include <graphene/custom_operations/custom_objects.hpp>
+#include <graphene/custom_operations/custom_operations.hpp>
+#include <graphene/custom_operations/custom_evaluators.hpp>
+
+namespace graphene { namespace custom_operations {
+using namespace chain;
+
+namespace detail
+{
+    class custom_operations_plugin_impl;
+}
+
+class custom_operations_plugin : public graphene::app::plugin
+{
+   public:
+      custom_operations_plugin();
+      virtual ~custom_operations_plugin();
+
+      std::string plugin_name()const override;
+      std::string plugin_description()const override;
+      virtual void plugin_set_program_options(
+         boost::program_options::options_description& cli,
+         boost::program_options::options_description& cfg) override;
+      virtual void plugin_initialize(const boost::program_options::variables_map& options) override;
+      virtual void plugin_startup() override;
+
+      friend class detail::custom_operations_plugin_impl;
+      std::unique_ptr<detail::custom_operations_plugin_impl> my;
+};
+
+typedef fc::static_variant<account_storage_map> custom_plugin_operation;
+
+struct custom_op_visitor
```

Comment:
This is an implementation detail, IMO it should be moved into the `.cpp` file.

Reply:
moved at https://github.com/bitshares/bitshares-core/pull/1926/commits/d381a7e5f8e39a7e3749576b09204485dca65789